tools/crm_rule.c

```diff
@@ -25,28 +25,60 @@ enum crm_rule_mode {
 
-static struct crm_option long_options[] = {
-    /* Top-level Options */
-    {"help",       no_argument,       NULL, '?', "\tThis text"},
-    {"version",    no_argument,       NULL, '$', "\tVersion information"  },
-    {"verbose",    no_argument,       NULL, 'V', "\tIncrease debug output"},
-
-    {"-spacer-",   required_argument, NULL, '-', "\nModes (mutually exclusive):" },
-    {"check",      no_argument,       NULL, 'c', "\tCheck whether a rule is in effect" },
-
-    {"-spacer-",   required_argument, NULL, '-', "\nAdditional options:" },
-    {"date",       required_argument, NULL, 'd', "Whether the rule is in effect on a given date" },
-    {"rule",       required_argument, NULL, 'r', "The ID of the rule to check" },
-
-    {"-spacer-",   no_argument,       NULL, '-', "\nData:"},
-    {"xml-text",   required_argument, NULL, 'X', "Use argument for XML (or stdin if '-')"},
-
-    {"-spacer-",   required_argument, NULL, '-', "\n\nThis tool is currently experimental.",
-     pcmk_option_paragraph},
-    {"-spacer-",   required_argument, NULL, '-', "The interface, behavior, and output may "
-                                                 "change with any version of pacemaker.", pcmk_option_paragraph},
-
-    {0, 0, 0, 0}
+static pcmk__cli_option_t long_options[] = {
+    // long option, argument type, storage, short option, description, flags
+    {
+        "help", no_argument, NULL, '?',
+        "\tThis text", pcmk__option_default
+    },
+    {
+        "version", no_argument, NULL, '$',
+        "\tVersion information", pcmk__option_default
+    },
+    {
+        "verbose", no_argument, NULL, 'V',
+        "\tIncrease debug output", pcmk__option_default
+    },
+    {
+        "-spacer-", no_argument, NULL, '-',
+        "\nModes (mutually exclusive):", pcmk__option_default
+    },
+    {
+        "check", no_argument, NULL, 'c',
+        "\tCheck whether a rule is in effect", pcmk__option_default
+    },
+    {
+        "-spacer-", no_argument, NULL, '-',
+        "\nAdditional options:", pcmk__option_default
+    },
+    {
+        "date", required_argument, NULL, 'd',
+        "Whether the rule is in effect on a given date", pcmk__option_default
+    },
+    {
+        "rule", required_argument, NULL, 'r',
+        "The ID of the rule to check", pcmk__option_default
+    },
+    {
+        "-spacer-", no_argument, NULL, '-',
+        "\nData:", pcmk__option_default
+    },
+    {
+        "xml-text", required_argument, NULL, 'X',
+        "Use argument for XML (or stdin if '-')", pcmk__option_default
+    },
+    {
+        "-spacer-", no_argument, NULL, '-',
+        "\n\nThis tool is currently experimental.",
+         pcmk__option_paragraph
+    },
+    {
+        "-spacer-", no_argument, NULL, '-',
+        "The interface, behavior, and output may change with any version of "
+            "pacemaker.",
+        pcmk__option_paragraph
+    },
+    { 0, 0, 0, 0 }
```

Comment:
Don't worry about changing crm_rule.  I have a couple patches I haven't pushed that'll convert crm_rule to using glib.  I'll push it as soon as #1982 gets merged.

Reply:
ok

Reply:
hmm actually then it wouldn't compile, so i'll leave it. in your checkout you can do:

git checkout 0ac587e34dcc27c6e06850149fcc30909918acb4~1 -- tools/crm_rule.c

and commit that to revert my changes before applying yours. When you're happy with it you can squash those commits so it looks like you edited my code :)